api/envoy/extensions/filters/http/router/v3/router.proto

```diff
@@ -89,4 +90,20 @@ message Router {
   // :ref:`gRPC stats filter<config_http_filters_grpc_stats>` documentation
   // for more details.
   bool suppress_grpc_request_failure_code_stats = 7;
+
+  // .. warning::
+  //   Upstream HTTP filters are not supported by default.
+  //   This warning will be removed as support moves beyond alpha.
+  //
+  // Optional HTTP filters for the upstream filter chain.
+  //
+  // These filters will be applied for all requests that pass through the router.
+  // They will also be applied to shadowed requests.
+  // Upstream filters cannot change route or cluster.
+  //
+  // If using upstream filters, please be aware that local errors sent by
+  // upstream filters will not trigger retries, and local errors sent by
```

Comment:
That's a question for Alyssa -- this is a copy of the documentation comment around upstream filters in upstream http protocol options.

Reply:
because we retry when the peer sends an error.  the upstream filter chain sendLocalReply kicks off the downstream sendLocalReply chain which bypasses the "should we retry" logic.